src/compile/legend/parse.ts

```diff
@@ -59,7 +59,7 @@ export function parseLegend(model: UnitModel, channel: Channel): VgLegend {
   }
 
   // 1.2 Add properties without rules
-  ['offset', 'orient'].forEach(function(property) {
+  ['maxExtent', 'offset', 'orient','titlePadding', 'zindex'].forEach(function(property) {
```

Comment:
I think you added more properties to legend.ts than you have added here?  
Also I don't see maxExtent below in Legend.